### PR TITLE
E2e testing: Add test for splitting/merging paragraph blocks with Enter/Backspace

### PR DESCRIPTION
**Pull request by twsp**
## Description
Added 005-splitting-merging.js, which currently performs an e2e test to ensure that Enter correctly splits paragraph blocks, and Backspace correctly merges them.

## How Has This Been Tested?
This script was tested using Jest and a version of Gutenberg running on localhost. The test was performed by running the command 'npx jest splitting-merging.test.js --watch'.  I compared the test's results to the results from manually splitting and merging paragraph blocks, and they were the same.

## Types of changes
This is a non-breaking change that adds an e2e test of basic functionality.

## Checklist:
- [X] My code is tested.
- [X] My code follows the WordPress code style.
- [X] My code has proper inline documentation.

### CODE REVIEW

File: test/e2e/specs/splitting-merging.test.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+		await page.keyboard.press( 'Enter' );
+
+		//Switch to Code Editor to check HTML output
+		await page.click( '.edit-post-more-menu [aria-label="More"]' );
+		let codeEditorButton = ( await page.$x( '//button[contains(text(), \'Code Editor\')]' ) )[ 0 ];
+		await codeEditorButton.click( 'button' );
+
+		//Assert that there are now two paragraph blocks with correct content
+		let textEditorContent = await page.$eval( '.editor-post-text-editor', ( element ) => element.value );
+		expect( textEditorContent ).toMatchSnapshot();
```
Comment (youknowriad) [inline test/e2e/specs/splitting-merging.test.js]: Not related to this PR, but I feel like we have a bunch of assertions like this:

 - Move to the text Editor
 - check the content with a snapshot
 - Move back to the visual editor

maybe we should create an utility for that?

**Review by youknowriad — APPROVED** [review]
Thanks @twsp for this PR. This is indeed a nice test to add. 👍

**Comment by gziolo** [comment]
@twsp 🎉 You made it, thanks again for bearing with all the recent changes in e2e area 💯

**Comment by twsp** [comment]
@gziolo No problem, overall I'm definitely liking the changes

**Comment by jeffpaul** [comment]
@twsp I've tried to reach out via other channels, but I have been unsuccessful in getting in touch. I wanted to make sure you saw the request to you for consent to re-license your contributions to Gutenberg under GPLv2 and MPLv2 here: https://github.com/WordPress/gutenberg/issues/31893. If you would kindly review that description and comment accordingly it would be greatly appreciated.

Thanks!
Jeff.

**Pull request merged**